src/Cli/dotnet/Parser.cs

```diff
@@ -217,7 +190,7 @@ public override void Write(HelpContext context)
                 }
                 else if (command.Name.Equals(NewCommandParser.GetCommand().Name))
                 {
-                    NewCommandShim.Run(context.ParseResult.GetArguments());
+                    NewCommandShim.Run(helpArgs);
```

Comment:
this fix is still needed. `dotnet new --help` and `dotnet new console --help` are different command.

Reply:
we dug into this some more - on its face this seems wrong for `dotnet new` (and any command with subcommands) but it turns out that in this revert the whole issue is sidestepped by the way the built-in commands are invoked in `Program.cs`. If a built-in command is detected then `Program.cs` forwards along the entire parseResult argument array to that command, which for `dotnet new` results in the correct help being returned, despite what this `if` block would suggest.  I've tested this branch locally to ensure this is the case, and we do get template-specific help when `dotnet.dll new console --help` is invoked.

Therefore, I suggest that we delete this `else if` entirely for clarity reasons on this 6.0.2xx branch. See the linked suggestion

Reply:
We checked it more closely with @baronfel, due to other changes in Program.cs this code is not used anymore. Any dotnet new command is redirected to `NewCommandShim` as it is part of `BuiltInCommand`. 

However, the code is still invalid. I would recommend to remove this if branch as once this code becomes reachable the bug discovered in https://github.com/dotnet/sdk/issues/23162 will come back.